scheduler/src/test/java/gov/nasa/jpl/aerie/scheduler/SimulationFacadeTest.java

```diff
@@ -188,55 +172,181 @@ public void associationToExistingSatisfyingActivity(){
+    final var actual = stateQuery.getValue(facade.getLatestConstraintSimulationResults(), null, horizon.getHor());
     assertThat(actual).isEqualTo(SerializedValue.of(3.0));
   }
 
   @Test
   public void getValueAtTimeDoubleOnSimplePlan() throws SimulationFacade.SimulationException {
     facade.simulateActivities(makeTestPlanP0B1().getActivities());
-    var actual = getFruitRes().getValueAtTime(t2);
+    facade.computeSimulationResultsUntil(tEnd);
+    final var stateQuery = new StateQueryParam(getFruitRes().name, new TimeExpressionConstant(t2));
+    final var actual = stateQuery.getValue(facade.getLatestConstraintSimulationResults(), null, horizon.getHor());
     assertThat(actual).isEqualTo(SerializedValue.of(2.9));
   }
 
   @Test
   public void whenValueAboveDoubleOnSimplePlan() throws SimulationFacade.SimulationException {
     facade.simulateActivities(makeTestPlanP0B1().getActivities());
-    var actual = getFruitRes().whenValueAbove(SerializedValue.of(2.9), entireHorizon);
+    facade.computeSimulationResultsUntil(tEnd);
+    var actual = new GreaterThan(getFruitRes(), new RealValue(2.9)).evaluate(facade.getLatestConstraintSimulationResults());
     var expected = new Windows(Window.betweenClosedOpen(t0, t2));
     assertThat(actual).isEqualTo(expected);
   }
 
   @Test
   public void whenValueBelowDoubleOnSimplePlan() throws SimulationFacade.SimulationException {
     facade.simulateActivities(makeTestPlanP0B1().getActivities());
-    var actual = getFruitRes().whenValueBelow(SerializedValue.of(3.0), entireHorizon);
-    var expected = new Windows(Window.betweenClosedOpen(t2, tEnd));
+    facade.computeSimulationResultsUntil(tEnd);
+    var actual = new LessThan(getFruitRes(), new RealValue(3.0)).evaluate(facade.getLatestConstraintSimulationResults());
+    var expected = new Windows(Window.between(t2, tEnd));
     assertThat(actual).isEqualTo(expected);
   }
 
   @Test
   public void whenValueBetweenDoubleOnSimplePlan() throws SimulationFacade.SimulationException {
     facade.simulateActivities(makeTestPlanP0B1().getActivities());
-    var actual = getFruitRes().whenValueBetween(SerializedValue.of(3.00), SerializedValue.of(3.99), entireHorizon);
+    facade.computeSimulationResultsUntil(tEnd);
+    var actual = new And(new GreaterThanOrEqual(getFruitRes(), new RealValue(3.0)), new LessThanOrEqual(getFruitRes(), new RealValue(3.99))).evaluate(facade.getLatestConstraintSimulationResults());
     var expected = new Windows(Window.betweenClosedOpen(t1, t2));
     assertThat(actual).isEqualTo(expected);
   }
 
   @Test
   public void whenValueEqualDoubleOnSimplePlan() throws SimulationFacade.SimulationException {
     facade.simulateActivities(makeTestPlanP0B1().getActivities());
-    var actual = getFruitRes().whenValueEqual(SerializedValue.of(3.00), entireHorizon);
+    facade.computeSimulationResultsUntil(tEnd);
+    var actual = new Equal(getFruitRes(), new RealValue(3.0)).evaluate(facade.getLatestConstraintSimulationResults());
```

Comment:
Very minor point - since `Equal` is generic over the profile type, we should always include `<>` when instantiating it:
```suggestion
    var actual = new Equal<>(getFruitRes(), new RealValue(3.0)).evaluate(facade.getLatestConstraintSimulationResults());
```

(EDIT: Joel made the same comment in his review)